x-pack/plugins/uptime/public/components/functional/__tests__/ping_list.test.tsx

```diff
@@ -0,0 +1,273 @@
+
+  beforeEach(() => {
+    pingList = {
+      allPings: {
+        total: 798,
+        pings: [
+          {
+            timestamp: '2019-01-18T22:02:31.383Z',
+            http: { response: { status_code: 200 } },
+            error: null,
+            monitor: {
+              duration: { us: 925862 },
+              id: 'http@https://www.elastic.co',
+              ip: '151.101.202.217',
+              name: 'http',
+              scheme: 'https',
+              status: 'up',
+              type: 'http',
+            },
+          },
+          {
+            timestamp: '2019-01-18T22:02:34.383Z',
+            http: { response: { status_code: 200 } },
+            error: null,
+            monitor: {
+              duration: { us: 738176 },
+              id: 'http@https://www.elastic.co',
+              ip: '151.101.202.217',
+              name: 'http',
+              scheme: 'https',
+              status: 'up',
+              type: 'http',
+            },
+          },
+          {
+            timestamp: '2019-01-18T22:02:37.383Z',
+            http: { response: { status_code: 200 } },
+            error: null,
+            monitor: {
+              duration: { us: 684949 },
+              id: 'http@https://www.elastic.co',
+              ip: '151.101.202.217',
+              name: 'http',
+              scheme: 'https',
+              status: 'up',
+              type: 'http',
+            },
+          },
+          {
+            timestamp: '2019-01-18T22:02:40.384Z',
+            http: { response: { status_code: 200 } },
+            error: null,
+            monitor: {
+              duration: { us: 773959 },
+              id: 'http@https://www.elastic.co',
+              ip: '151.101.202.217',
+              name: 'http',
+              scheme: 'https',
+              status: 'up',
+              type: 'http',
+            },
+          },
+          {
+            timestamp: '2019-01-18T22:02:43.385Z',
+            http: { response: { status_code: 200 } },
+            error: null,
+            monitor: {
+              duration: { us: 483803 },
+              id: 'http@https://www.elastic.co',
+              ip: '151.101.202.217',
+              name: 'http',
+              scheme: 'https',
+              status: 'up',
+              type: 'http',
+            },
+          },
+        ],
+      },
+    };
+  });
+
+  it('renders sorted list without errors', () => {
+    const { allPings } = pingList;
+    const component = shallowWithIntl(
+      <PingList
+        loading={false}
+        maxSearchSize={200}
+        pingResults={allPings}
+        searchSizeOnBlur={jest.fn()}
+        selectedOption={{ label: 'All', value: '' }}
+        selectedOptionChanged={jest.fn()}
+        statusOptions={[{ label: 'All', value: '' }]}
+      />
+    );
+    expect(component).toMatchSnapshot();
+  });
+
+  it('renders unsorted list of many monitors without errors', () => {
+    pingList = {
```

Comment:
Is there a way we can ensure that this variable stays in sync with the GQL definitions of this structure? Isn't there a type we can set here instead of `any` that would cause this test to fail on schema change?

Reply:
Nevermind, the component type checks.